### ISSUE TITLE
False negative `checked_conversions` for Int::MAX

### ISSUE THREAD

**Issue by lzutao** [rust-lang/rust-clippy#5675]
Example (playground: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=f64cf36f94d0d7480c40454b89ffc06e)
```rust
#![warn(clippy::checked_conversions)]

pub fn foo(value: i64) -> Option<u32> {
    let _ = value <= (u32::max_value() as i64) && value >= 0;
    //^WARN: Checked cast can be simplified.
    let _ = value <= (u32::MAX as i64) && value >= 0; // This should also be warned but it doesn't
    Some(42)
}
```
